Initial Commit

rodney528 opened the pull request:
In difficultyArray you can put a whole new difficulty list which will fix some issues when changing songs doesn't want to fucking work! Because if that song doesn't have said chart/diff then nothing happens so PLZ MERGE THIS!!

charlesisfeline reviewed (approved):
but mistakes aside, this is, pretty nice

jimmyjamescool commented:
I HATE MAPS. I FUCKING WANNA BURN EVERY SINGLE MAP I SEE. YOU LOST? GET LOST. HAVE A DEGREE IN CARTOGRAPHY? GO FUCK YOURSELF. YOU WANNA BE A PIRATE AND SAIL THE SEVEN SEAS? GO TAKE A PIPE BOMB WITH YOU! LIKE BEING ABLE TO TAKE THE FASTEST ROUTE TO AND FROM WORK? I'M GOING TO KILL YOU. YOU FUCKING KNOW WHAT MAPS ARE? JUMP OFF A CLIFF! EVERY MAP LOVER, HATER, USER, AND THE LIKE; ALL SHALL MEET THEIR DOOM!

TehPuertoRicanSpartan commented:
what the fuck happened

RodneyAnImaginativePerson commented:
> you forgot to change the pr name tho

I deleted the old one, thats why I kept the name the same.

RodneyAnImaginativePerson commented on source/psychlua/FunkinLua.hx:
```diff
@@ -419,11 +419,10 @@ class FunkinLua {
-			if (difficultyNum == -1)
-				difficultyNum = PlayState.storyDifficulty;
+		Lua_helper.add_callback(lua, "loadSong", function(?name:String = null, ?difficultyNum:Int = -1, ?difficultyArray:Array<String> = null) {
+			if (difficultyArray != null) CoolUtil.difficulties = difficultyArray;
```
Woops

RodneyAnImaginativePerson replied:
0.6.3 habits

RodneyAnImaginativePerson commented:
poop to bigAssShit lmao

ShadowMario committed:
Credit removed by choice of the contributor

ShadowMario committed:
V-Slice Importer/Exporter and adjustments

ShadowMario committed:
fix crash with no save data

crowplexus committed:
Update readme.txt

crowplexus committed:
Update README.md

crowplexus committed:
Update CreditsState.hx

crowplexus committed:
Update DialogueBoxPsych.hx

ShadowMario committed:
Minor fixes

ShadowMario committed:
fixed hitsound when passing events

ShadowMario committed:
Merge pull request #15108 from crowplexus/patch-shubs

Remove any remaining references to Yoshubs from the source code

ShadowMario committed:
Merge pull request #14919 from ReXiSp/experimental

Fix deleting gamepad bindings

ShadowMario committed:
weekend 1 port (unfinished, dont complain)

ShadowMario committed:
forgot to push weeks list

ShadowMario committed:
blazin stage didnt push properly

crowplexus committed:
lock library versions in setup files

crowplexus committed:
Delete hmm.json

crowplexus committed:
hopefully fix gh actions

crowplexus committed:
Update source/Main.hx

crowplexus committed:
fix stress cutscene crash

ShadowMario committed:
Merge pull request #15154 from crowplexus/lock-libs

lock library versions in setup files

ShadowMario committed:
flxanimate fix

ShadowMario committed:
Revert "flxanimate fix"

This reverts commit 16edae7412e33959bd8a2293342e908a695cc1d6.

ShadowMario committed:
Weekend1 Cutscenes + Minor fixes/optimization

ShadowMario committed:
Fixed section BPM grid line position

ShadowMario committed:
Move notes and audio offset completed

ShadowMario committed:
Chart editor autosave system done

ShadowMario committed:
Compilation fix and skippable cutscenes/dialogues

ShadowMario committed:
Metronome and "Go to" done

tposejank committed:
Fix deprecated actions warnings

Fixes deprecated actions versions that will stop working soon.

ShadowMario committed:
Copy and Paste + Vortex Editor

ShadowMario committed:
Chart Editor Undo/Redo (Ctrl + Z/Y) and Tip text

ShadowMario committed:
Merge pull request #15212 from tposejank/fix-deprecation-warnings

Fix deprecated actions warnings

ShadowMario committed:
Merge pull request #14892 from CrusherNotDrip/freeplay-color

Remove the `freeplayColors` array

ShadowMario committed:
Ctrl X, translatable credits and mod settings

ShadowMario committed:
Fixed NoteSplash antialiasing

ShadowMario committed:
Controls null checking

crowplexus committed:
clear ghost notes!!!

crowplexus committed:
layer notes over HUD

crowplexus committed:
Merge branch 'experimental' into input-fix

crowplexus committed:
Update Note.hx

crowplexus committed:
Update PlayState.hx

crowplexus committed:
Update EditorPlayState.hx

crowplexus committed:
Update Note.hx

crowplexus committed:
Update PlayState.hx

crowplexus committed:
Update EditorPlayState.hx

crowplexus committed:
Update Note.hx

ShadowMario committed:
Merge pull request #15227 from crowplexus/patch-1

Make a variable so people can force hitsounds to be played in note types

ShadowMario committed:
Fixed typo on get_hitsoundVolume

ShadowMario committed:
Merge pull request #14818 from richTrash21/experimental-hscript-return-fix

[BUGFIX] Hscript early return fix

tposejank committed:
building guide grammar fixes

markdown syntax fixes
removed words like "recommended," "may," "going to want," "pretty much," which do not explicitly state a requirement
unix setup cd command added
added lime test platform -clean error guide.

tposejank committed:
enable pretty message reporting on newer haxe vers

plus a hxcpp debug server fix ("60").

tposejank committed:
change wiki link to a redirect

tposejank committed:
these commands were there

tposejank committed:
add back this

tposejank committed:
cpp instead of platform

ShadowMario committed:
Weekend 1 Game overs

ShadowMario committed:
Merge branch 'experimental' of https://github.com/ShadowMario/FNF-PsychEngine into experimental

ShadowMario committed:
Update Pico_Death.png

ShadowMario committed:
Merge pull request #15224 from crowplexus/input-fix

Input Patch - No More Ghost Notes!

ShadowMario committed:
Chart editor bugfix and vortex row lines

ShadowMario committed:
Update Main.hx

ShadowMario committed:
PsychUI Bugfixes, notetypes.txt for blazin/2hot

ShadowMario committed:
Fixed missing character on character editor

ShadowMario committed:
Workaround for stage editor selection box lag

ThomasDarkson committed:
note splash editor

ShadowMario committed:
Merge pull request #15239 from TahirToprakKarabekiroglu/experimental

note splash editor

ShadowMario committed:
Merge pull request #15229 from tposejank/fix-docs

Building documentation fixes plus some additions

melodiekit committed:
fix bools on set property event

ShadowMario committed:
I forgot about LBRACKET/RBRACKET

Unholywanderer committed:
Update tutorial-easy.json

Unholywanderer committed:
Update tutorial.json

Unholywanderer committed:
Update tutorial-hard.json

ShadowMario committed:
Merge pull request #15244 from Unholywanderer/experimental

let Tutorial actually play the voices

ShadowMario committed:
Merge pull request #15241 from LarryFrosty/pro

Bool value fix for "Set Property" event

ShadowMario committed:
Merge pull request #14403 from LarryFrosty/experimental

Fix onMoveCamera

HomuHomu833 committed:
Update FPSCounter.hx

HomuHomu833 committed:
or actually

ShadowMario committed:
Merge pull request #15246 from MobilePorting/patch-1

switch `cast(System.totalMemory, UInt)` to use `cpp.vm.Gc.memInfo64(cpp.vm.Gc.MEM_INFO_USAGE)`

ShadowMario committed:
Discord presence to Stage/Chart Editor

ShadowMario committed:
Update NoteSplashEditorState.hx

melodiekit committed:
fix varsToBring

melodiekit committed:
i forgot

DM-kun committed:
some improvements

DM-kun committed:
fixed "affected by rgb" option

DM-kun committed:
formatting + more improvs

DM-kun committed:
changed "affectedByShader" to "allowRGB" and added "allowPixel"

DM-kun committed:
fix checkbox offset

melodiekit committed:
Fix onGhostTap

ShadowMario committed:
Merge pull request #15250 from LarryFrosty/hscript

Use a setter for `varsToBring`

melodiekit committed:
fix 2

melodiekit committed:
OOPS

ShadowMario committed:
Stage Editor loading screen preload config

ShadowMario committed:
Fixed Loading of files in week_assets

ShadowMario committed:
Update PlayState.hx

ShadowMario committed:
Prerelease build in progress

ShadowMario committed:
Merge pull request #15275 from LarryFrosty/0.694267515923567

Fix onGhostTap

ShadowMario committed:
Changed SScript version

ShadowMario committed:
Added modFolder variable to lua

tposejank committed:
Update windows.bat

tposejank committed:
Update unix.sh

ShadowMario committed:
Merge pull request #15317 from tposejank/yea

Yea (Fix experimental compilation)

ShadowMario committed:
Merge pull request #15255 from DMMaster636/experimental

Note Splash Editor Improvements

ShadowMario committed:
Update HScript.hx

ShadowMario committed:
Merge branch 'experimental' of https://github.com/ShadowMario/FNF-PsychEngine into experimental

ShadowMario committed:
Stage JSON images automatically preload

ShadowMario committed:
Removed lastCharFocus

ShadowMario committed:
crash preventions

ShadowMario committed:
Changed HScript version (for now)

ShadowMario committed:
Fixed missing shot sound on Weekend 1 Story

ShadowMario committed:
null checks to try and prevent a crash

melodiekit committed:
allow cameraFromString to use custom cameras

melodiekit committed:
make setProperty return the value

melodiekit committed:
make these damn variables actually work

Fusion243 committed:
Fixed a couple bugs i found

- Put the camPos var outside of the lua/hscript stuff to fix a compiling error when they are both disabled.
- Moved persistentUpdate above the weeksList.length check to stop it from setting to true even if there are no weeks loaded.

betpowo committed:
images from assets folder in stage editor (???)

there was an oopsie

crowplexus committed:
Update PlayState.hx

crowplexus committed:
Update PlayState.hx

crowplexus committed:
Update EditorPlayState.hx

crowplexus committed:
if i can spell to save my life

ShadowMario committed:
Merge pull request #15338 from crowplexus/patch-1

someone asked me to fix these even though its a massive edge case that might never happen

ShadowMario committed:
Merge pull request #15332 from LarryFrosty/cameraFromString

allow cameraFromString to use custom cameras

ShadowMario committed:
Merge pull request #15337 from betpowo/patch-7

images from assets folder in stage editor (???)

ShadowMario committed:
Merge pull request #15336 from Fusion243/experimental

Fixed a couple bugs i found in prerelease

ShadowMario committed:
Changed SScript to HScript-Iris on authors request

ShadowMario committed:
Merge branch 'experimental' of https://github.com/ShadowMario/FNF-PsychEngine into experimental

ShadowMario committed:
R to reset section

ShadowMario committed:
Changed botplay text position

Sword352 committed:
restore the old behaviour for setting variables to hscript

ShadowMario committed:
Merge pull request #15354 from Sword352/hscript-set-bugfix

Restore the old behaviour for setting variables to hscript

melodiekit committed:
One

melodiekit committed:
Update T

melodiekit committed:
Lua :pensive:

melodiekit committed:
FlixelText.hx

lemz1 committed:
fix async loading

crowplexus committed:
Update StoryMenuState.hx

crowplexus committed:
replace all occurences of VCR OSD Mono with Paths.font("vcr.ttf")

ShadowMario committed:
Update funkin.vis to compile!

lemz1 committed:
some cleanup

Fearester2008 committed:
"New" Button Crash Fix in Chart Editor!

100% work

melodiekit committed:
Fix Game Over music not looping

melodiekit committed:
Remove text arg from getLuaObject

ShadowMario committed:
Added missing credit icons and achievements

ShadowMario committed:
Merge pull request #15331 from LarryFrosty/prop-and-substate

Make setProperty return the value and fix custom substate hscript variables

ShadowMario committed:
Merge pull request #15428 from LarryFrosty/bend-ova

Fix Game Over music not looping

richTrash21 committed:
use hxdiscord_rpc 1.2.1

crowplexus committed:
Update FPSCounter.hx

melodiekit committed:
change type of Function_Stop and variations to String

melodiekit committed:
part 2

crowplexus committed:
fix fps counting

crowplexus committed:
Update HealthIcon.hx

melodiekit committed:
Change BaseStage.Countdown to Countdown in Template.hx

ACoolioDude committed:
Adding the command to make directory for haxelib and setup it in same time for Unix systems

- For those who are lazy to make manually folder and setup haxelib in same time, they need this

melodiekit committed:
Add missing translations check

melodiekit committed:
nvm

melodiekit committed:
BRUH I WAS LOOKING IN THE WRONG PLACE

melodiekit committed:
fix stopSound, pauseSound, resumeSound

melodiekit committed:
brah

ShadowMario committed:
Fixed compiling without BASE_GAME_FILES

ShadowMario committed:
Merge pull request #15509 from LarryFrosty/what-do-i-put-here

Add missing `TRANSLATIONS_ALLOWED` check

ShadowMario committed:
Merge pull request #15507 from Stefan2008Git/patch-2

Adding the command to make directory for haxelib and setup it in same…

ShadowMario committed:
Merge branch 'experimental' into discord-rpc-fix

ShadowMario committed:
Update unix.sh

ShadowMario committed:
Update windows.bat

ShadowMario committed:
Merge pull request #15482 from richTrash21/discord-rpc-fix

Upgrade hxdiscord_rpc

ShadowMario committed:
Merge pull request #15497 from crowplexus/patch-7

allow health icons to have infinite sizes

ShadowMario committed:
Merge pull request #15399 from crowplexus/patch-1

Use modded fonts in Story Mode Menu and other missing places

ShadowMario committed:
Updated linc_luajit, hxvlc and hscript-iris

ShadowMario committed:
Fixed "Tried to call a non-function" errors

ShadowMario committed:
Merge pull request #15483 from crowplexus/patch-6

Fix overlay updating every frame instead of every second

ShadowMario committed:
Merge pull request #15487 from LarryFrosty/idk

Change the type of `Function_Stop` and its variations to `String`

ShadowMario committed:
Merge pull request #15418 from Fearester2008/patch-5

"New" Button Crash Fix in Chart Editor!

ShadowMario committed:
Merge pull request #15386 from lemz1/fix-loading-screen

Fix Loading State

ShadowMario committed:
Fixed modded files preload and health icon sizes

ShadowMario committed:
Merge pull request #15529 from LarryFrosty/bruh

Fix `stopSound`, `pauseSound`, and `resumeSound`

ShadowMario committed:
Minor chart editor fixes

ShadowMario committed:
Merge branch 'experimental' of https://github.com/ShadowMario/FNF-PsychEngine into experimental

ShadowMario committed:
Brazil update and fixed credits links

ShadowMario committed:
New Riveren and Pessy title bumps

ShadowMario committed:
Fixed GF section not working for opponent

afeefflox committed:
Fix V-Silce

idk if the first bpm didn't exist it might cause skip on start song
and remove gf (idk sometime people like without gf)

glowsoony committed:
Fixes Inst/Voices Loading For Some Reason

Bug causes freakyMenu to play because it couldn't find inst or voices

ShadowMario committed:
Merge pull request #15555 from afeefflox/patch-12

Fix V-Silce

ShadowMario committed:
oh my fucking god i finally fixed this AHH MY BRAIN

ShadowMario committed:
F5 on Pause to reload all and Paths.getMultiAtlas

melodiekit committed:
close custom substate on destroy

should prevent crashing if the state was changed or reset while the substate is still active

melodiekit committed:
actual fix LOOOOL

melodiekit committed:
fuck this bullshit

melodiekit committed:
set instance to null on destroy

melodiekit committed:
remove that

melodiekit committed:
go back to old fix

melodiekit committed:
actually this isnt even necessary idk why i did that

melodiekit committed:
Certified Dumbass

ShadowMario committed:
Renamed NotesSubState to NotesColorSubState

crowplexus committed:
Update PlayState.hx

crowplexus committed:
Update FunkinLua.hx

ShadowMario committed:
Merge pull request #15580 from crowplexus/patch-1

set totalPlayed and totalNotesHit on scripts

ShadowMario committed:
Fixed sustain length being wrong when changing BPM

ShadowMario committed:
FLA upload and little tweaks

F5 also reloads translations now too

ShadowMario committed:
Alternative to unlock pessy achievement

ShadowMario committed:
Swapped around all "\\" with "/" on chart editor

ShadowMario committed:
Fixed stage editor not loading from paths with "."

ShadowMario committed:
Update ModsMenuState.hx

ShadowMario committed:
Added 8 new camera functions to lua

ShadowMario committed:
Wrong branch!

ShadowMario committed:
Added fadeOut to cameraFade function

ShadowMario committed:
minor tweak on achievements and lua death counter

ShadowMario committed:
Minor changes and removed broken functions

ShadowMario committed:
Reverted changes in some lua functions and deprecated addHaxeLibrary

ShadowMario committed:
Improvements for some Lua functions and tweaks

melodiekit committed:
Merge branch 'experimental' into testing

ShadowMario committed:
Added "group" for (set/get)ObjectOrder

ShadowMario committed:
minor changes

ShadowMario committed:
instanceArg for setProperty functions option

PartnerCapPikachu committed:
Update sustains to BPM they spawn in

PartnerCapPikachu committed:
fix inconsitent sustain length

PartnerCapPikachu committed:
nvm it just broke other sustains

PartnerCapPikachu committed:
NOW it should fix inconsitent sustains without breaking them on other bpms

PartnerCapPikachu committed:
small fix and round up a tail to whole length

example:
sustain length of one step and a half would round up to two steps instead

PartnerCapPikachu committed:
roudn down instead

ShadowMario committed:
updateHitbox should be working properly now

PartnerCapPikachu committed:
fix resetAnim and sync it with playbackRate

ShadowMario committed:
Updated some lua functions

ShadowMario committed:
Fixed issues and finished To Do in stage editor

ShadowMario committed:
Update Song.hx

ShadowMario committed:
More lua function fixes

melodiekit committed:
fix instanceArg second argument

ShadowMario committed:
update sustain length visual when BPM is changed

ShadowMario committed:
Merge pull request #15627 from PartnerCapPikachu/experimental

Update sustains to BPM they spawn in

ShadowMario committed:
Update ChartingState.hx

ShadowMario committed:
Merge pull request #15653 from LarryFrosty/he

Fix instanceArg second argument

melodiekit committed:
Merge branch 'experimental' into testing

ShadowMario committed:
Changed text alignment in Controls substate

ShadowMario committed:
Update pt-BR.lang

ShadowMario committed:
Merge pull request #15360 from LarryFrosty/testing

Small improvements

ShadowMario committed:
Merge pull request #15572 from LarryFrosty/substate-the-sequel

Close custom substate on destroy

ShadowMario committed:
Merge pull request #15558 from glowsoony/experimental

Fix Looking for soundFile when preloading. (Experimental Branch)

ShadowMario committed:
Splashes demo in Visual Settings

ShadowMario committed:
Nearly last fixes

ShadowMario committed:
Fixed some issues found by LarryFrosty

ShadowMario committed:
Readme and hxvlc tweaks

ShadowMario committed:
Update ModsMenuState.hx

ShadowMario committed:
Should fix the weird controller issue

ShadowMario committed:
Changed sustain spawning once again

ShadowMario committed:
FPSCounter updates more times per second now

ShadowMario committed:
Corrected resetAnim -Pumpsuki

ShadowMario committed:
Fixed themes bugging charter and added new theme

ShadowMario committed:
Fixed compiling with some different settings

rodney528 committed:
Merge remote-tracking branch 'upstream/experimental' into loadSong-FIX

rodney528 committed:
Update FunkinLua.hx

rodney528 committed:
Update FunkinLua.hx
